benchmark.go

```diff
@@ -188,9 +214,75 @@ func runBenchmark(out output, logger *zap.Logger, allOpts Options, resolved reso
 		zap.Time("startTime", start),
 	)
 
+	if opts.Format == "json" || opts.Format == "JSON" {
+		outputJSON(opts, overall, out, goMaxProcs, numConns, total)
+	} else {
+		outputPlaintext(opts, overall, out, goMaxProcs, numConns, total)
+	}
+}
+
+// JSON output helper method
+func outputJSON(opts BenchmarkOptions, overall *benchmarkState, out output, goMaxProcs int, numConns int, total time.Duration) {
+	// Print out errors
 	overall.printErrors(out)
-	overall.printLatencies(out)
 
+	// Create Latencies, BenchmarkParameters, Summary, and BenchmarkOutput structs
+	latencyValues := overall.getLatencies(out)
+	benchmarkParams := BenchmarkParameters{
+		CPUs:        goMaxProcs,
+		Connections: numConns,
+		Concurrency: opts.Concurrency,
+		MaxRequests: opts.MaxRequests,
+		MaxDuration: opts.MaxDuration.String(),
+		MaxRPS:      opts.RPS,
+	}
+	latenciesOutput := Latencies{
+		P5000: latencyValues[0],
+		P9000: latencyValues[1],
+		P9500: latencyValues[2],
+		P9900: latencyValues[3],
+		P9990: latencyValues[4],
+		P9995: latencyValues[5],
+		P1000: latencyValues[6],
+	}
+	summary := Summary{
+		ElapsedTime:   (total / time.Millisecond * time.Millisecond).String(),
```

Comment:
it looks like there are quite some duplicated logic in `outputJSON` and `outputPlainText` (cacluating ElapsedTime/RPS). An idea: we always populate the BenchmarkParams/BenchmarkOutput structs before checking for `Format`, and then passing those structs to `outputJSON` and `outputPlainText` so that they can just use the results as is, without having to do the calculating again.